notebooks/1.processing-profiles/nbconverted/1.preprocessing-profiles.py

```diff
@@ -0,0 +1,205 @@
+
+# getting all platemaps files
+all_platemap_paths = list(metadata_dir.glob("Target_Selective_Library_Screen_*.csv"))
+
+# setting up results directory
+results_dir = pathlib.Path("./results").resolve()
+results_dir.mkdir(exist_ok=True)
+
+# making a results dir for the augmented platemaps with moa
+results_platemap_dir = (results_dir / "platemaps_w_moa").resolve()
+results_platemap_dir.mkdir(exist_ok=True)
+
+# making a results dir for aggregate profiles
+results_moa_agg_profiles_dir = (results_dir / "aggregated_profiles/moa_agg").resolve()
+results_repl_agg_profiles_dir = (results_dir / "aggregated_profiles/repl_agg").resolve()
+results_moa_agg_profiles_dir.mkdir(exist_ok=True, parents=True)
+results_repl_agg_profiles_dir.mkdir(exist_ok=True, parents=True)
+
+# updated barcode out path# Save the updated barcode metadata to a CSV file
+updated_barcode_path = (
+    results_platemap_dir / f"barcode_platemaps_{moa_tag}.csv"
+).resolve()
+
+
+# This process integrates pathway metadata into experimental platemaps to augment single-cell image-based profiles with additional biological context. First, the pathway metadata, containing information such as treatment identifiers, well positions, and pathways, is filtered for each specific plate. This filtered metadata is then merged with the corresponding platemap, linking treatments and well positions to their associated pathways. The resulting augmented platemaps provide a unified view that connects experimental treatments to mechanisms of action (MOA).
+
+# In[4]:
+
+
+# Load the pathways metadata into a DataFrame
+pathway_meta_df = pd.read_csv(pathway_metadata_path)
+
+# Iterate over all platemap files to augment them with pathway MOA information
+for platemap_path in all_platemap_paths:
+    # Generate an output filename for the augmented platemap
+    # Example: Extract prefix and plate ID, then append a "with_moa" tag
+    prefix = platemap_path.stem.rsplit("_", 2)[0]
+    plate_id = platemap_path.stem.split("_", 4)[-1]
+    tag = "with_moa"
+    outname = f"{prefix}_{plate_id}_{tag}.csv"
+
+    # Load the current platemap into a DataFrame
+    plate_meta_df = pd.read_csv(platemap_path)
+
+    # Filter the pathway metadata to include only rows corresponding to the current plate
+    # Select relevant columns: 'UCD ID' (treatment), 'Well' (well position), and 'Pathway' (moa)
+    plate_pathway_meta_df = pathway_meta_df.loc[pathway_meta_df["Plate"] == plate_id][
+        ["UCD ID", "Well", "Pathway"]
+    ]
+
+    # Merge the pathway metadata into the platemap using 'treatment' and 'well_position'
+    # as keys. Perform a left join to retain all rows from the platemap
+    merged_df = plate_meta_df.merge(
+        plate_pathway_meta_df,
+        left_on=["treatment", "well_position"],
+        right_on=["UCD ID", "Well"],
+        how="left",
+    )
+
+    # Save the augmented platemap with MOA information to the output directory
+    merged_df.to_csv(results_platemap_dir / outname, index=False)
```

Comment:
I could be wrong, but don't you already do this in the previous notebook? Recommend DNR and just loading in the resulting platemaps created in the previous notebook.